src/test/java/org/sagebionetworks/bridge/exporter/record/RecordIdSourceFactoryTest.java

```diff
@@ -27,96 +27,78 @@
                 .thenReturn(barStudyItemList);
 
-        ArgumentCaptor<RangeKeyCondition> fooRangeKeyCaptor = ArgumentCaptor.forClass(RangeKeyCondition.class);
-        when(mockQueryHelper.query(same(mockRecordIndex), eq("studyId"), eq("foo-study"), fooRangeKeyCaptor.capture()))
+        when(mockQueryHelper.query(same(mockRecordIndex), eq(STUDY_ID), eq("ddb-foo"), fooRangeKeyCaptor.capture()))
                 .thenReturn(fooStudyItemList);
 
         // set up factory
         RecordIdSourceFactory factory = new RecordIdSourceFactory();
-        factory.setConfig(mockConfig());
         factory.setDdbQueryHelper(mockQueryHelper);
+        factory.setConfig(mockConfig());
         factory.setDdbRecordStudyUploadedOnIndex(mockRecordIndex);
 
-        // finish building request - Use a TreeSet for the study whitelist, so we can get a deterministic test.
-        // (ImmutableSet preserves the iteration order.)
-        Set<String> studyWhitelist = new TreeSet<>();
-        studyWhitelist.add("bar-study");
-        studyWhitelist.add("foo-study");
-        BridgeExporterRequest request = requestBuilder.withStudyWhitelist(studyWhitelist).build();
-
         // execute and validate
-        Iterable<String> recordIdIter = factory.getRecordSourceForRequest(request);
+        BridgeExporterRequest request;
+        request = new BridgeExporterRequest.Builder()
+                .withEndDateTime(UPLOAD_END_DATE_TIME_OBJ)
+                .withExportType(ExportType.DAILY)
+                .build();
+
+        Iterable<String> recordIdIter = factory.getRecordSourceForRequest(request, UPLOAD_END_DATE_TIME_OBJ, studyIdsToQuery);
+
         List<String> recordIdList = ImmutableList.copyOf(recordIdIter);
-        assertEquals(recordIdList.size(), 4);
-        assertEquals(recordIdList.get(0), "bar-1");
-        assertEquals(recordIdList.get(1), "bar-2");
-        assertEquals(recordIdList.get(2), "foo-1");
-        assertEquals(recordIdList.get(3), "foo-2");
-
-        // validate range key queries
-        validateRangeKey(barRangeKeyCaptor.getValue(), expectedStartMillis, expectedEndMillis);
-        validateRangeKey(fooRangeKeyCaptor.getValue(), expectedStartMillis, expectedEndMillis);
+
+        if (!isEndDateTimeBeforeLastExportTime) {
+            assertEquals(recordIdList.size(), 4); // only output records in given time range
+            assertEquals(recordIdList.get(0), "foo-1");
+            assertEquals(recordIdList.get(1), "foo-2");
+            assertEquals(recordIdList.get(2), "bar-1");
+            assertEquals(recordIdList.get(3), "bar-2");
+
+            validateRangeKey(fooRangeKeyCaptor.getValue(), UPLOAD_START_DATE_TIME_OBJ.getMillis(), UPLOAD_END_DATE_TIME_OBJ.getMillis());
+            validateRangeKey(barRangeKeyCaptor.getValue(), UPLOAD_START_DATE_TIME_OBJ.getMillis(), UPLOAD_END_DATE_TIME_OBJ.getMillis());
+        } else {
+            assertEquals(recordIdList.size(), 4); // only output records in given time range
+        }
```

Comment:
yeah you are right. and i think we should test end date time before last export date time in the method bootstrapStudyIdsToQuery() instead of this -- since we move such logic in DynamoHelper class already.